File: Leetcode/3151. Special Array I.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        p=nums[0] & 1
        n=len(nums)
        for i in range(1,n):
            if (nums[i] & 1) == p:
                return False
            else:
                p=p ^ 1
        return True
        

File: Leetcode/3151. Special Array I.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int n=nums.size();
        int p=nums[0]&1;
        for(int i=1;i<n;i++){
            if((nums[i]&1) == p){
                return false;
            }else{
                p^=1;
            }
        }
        return true;
    }
};
